samples/rocdecDecode/rocdecdecode.cpp

```diff
@@ -0,0 +1,752 @@
+                            file_names_sub_folder.push_back(sub_entry.path());
+                        }
+                        std::sort(file_names_sub_folder.begin(), file_names_sub_folder.end(), compareFilenames);
+                        input_file_names.insert(input_file_names.end(), file_names_sub_folder.begin(), file_names_sub_folder.end());
+                        file_names_sub_folder.clear();
+                    } else if(entry.is_regular_file()) {
+                        b_sort_filenames = true;
+                        input_file_names.push_back(entry.path());
+                    }
+                    else {
+                        std::cout << "unknown file type in input folder: " << entry.path().string() << '\n';
+                        continue;
+                    }
+                 }
+                 if (b_sort_filenames) {
+                    std::sort(input_file_names.begin(), input_file_names.end(), compareFilenames);
+                 }
+            } else {
+                input_file_names.push_back(input_file_path);
+            }
+        
+            std::cout << "Read " << input_file_names.size() << " frames from disk." << std::endl;
+            continue;
+        }
+        if (!strcmp(argv[i], "-o")) {
+            if (++i == argc) {
+                ShowHelpAndExit("-o");
+            }
+            output_file_path = argv[i];
+            dec_info.output_file_path = output_file_path;
+            dump_output_frames = true;
+            continue;
+        }
+        if (!strcmp(argv[i], "-b")) {
+            if (++i == argc) {
+                ShowHelpAndExit("-b");
+            }
+            backend = static_cast<DecoderBackend>(atoi(argv[i]));
+            continue;
+        }
+
+        if (!strcmp(argv[i], "-d")) {
+            if (++i == argc) {
+                ShowHelpAndExit("-d");
+            }
+            device_id = atoi(argv[i]);
+            continue;
+        }
+        if (!strcmp(argv[i], "-c")) {
+            if (++i == argc) {
+                ShowHelpAndExit("-c");
+            }
+            codec_type = atoi(argv[i]);
+            continue;
+        }
+
+        if (!strcmp(argv[i], "-n")) {
+            if (++i == argc) {
+                ShowHelpAndExit("-n");
+            }
+            num_iterations = atoi(argv[i]);
+            continue;
+        }
+        ShowHelpAndExit(argv[i]);
+    }
+    dec_info.rocdec_codec_id = CodecTypeToRocDecVideoCodec(codec_type);
+    dec_info.dec_device_id = device_id;
+    dec_info.mem_type = (!backend) ? OUT_SURFACE_MEM_DEV_INTERNAL : OUT_SURFACE_MEM_HOST;
+    init();
+    if (backend == DECODER_BACKEND_DEVICE) {
+        create_parser(dec_info);
+        create_decoder(dec_info);
+    } else {
+        create_decoder_host(dec_info);
+    }
+    dec_info.dump_decoded_frames = dump_output_frames;
+    auto input_frames = read_frames(input_file_names);
+    decode_frames(dec_info, input_frames);  // warmup
```

Comment:
Please remove this line. The app provides `-n` for iterations, so warmup is not needed.